src/pages/Sidebar/HappinessForm.tsx

```diff
@@ -0,0 +1,213 @@
+import { useState, useRef, useEffect } from "react";
+import { useQuery, useMutation } from "react-query";
+import { useApi } from "../../contexts/ApiProvider";
+import { Happiness, NewHappiness } from "../../data/models/Happiness";
+import { validateHappiness, formatDate, formatHappinessNum } from "../../utils";
+import TextareaAutosize from "react-textarea-autosize";
+import HappinessNumber from "../../components/HappinessNumber";
+
+export default function HappinessForm({ height }: { height: number }) {
+  const { api } = useApi();
+
+  const [comment, setComment] = useState("");
+  const [submissionStatus, setSubmissionStatus] = useState("Updated");
+  // TODO refactor to use useRef! the let def was being reassigned
+  // on every recomposition
+  const postHappinessTimeout = useRef<number | undefined>(undefined);
+  const isInitialRender = useRef(true);
+
+  const [radioValue, setRadioValue] = useState(2);
+  const [selDate, setSelDate] = useState(new Date());
+  const [happiness, setHappiness] = useState(-1);
+
+  const UNSUBMITTED = "Unsubmitted (enter number to submit)";
+  const UPDATING = "Updating...";
+  const UPDATED = "Updated";
+  const ERROR = "Error loading/retrieving happiness";
+
+  const postHappinessMutation = useMutation((newHappiness: NewHappiness) =>
+    api.post("/happiness/", newHappiness),
+  );
+
+  useEffect(() => {
```

Comment:
Add a comment at the top of each `useEffect` describing what it's responsible for.